Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptSearch.java

```diff
@@ -97,14 +99,16 @@
             totalNumberOfRecords = manager.getTotalNumberOfRecordsForSearch(searchFields, operator);
             searchResults = manager.searchForConceptByPageNumberAndFieldMap(searchFields, operator,
                     page, numberOfRecordsPerPage);
-        } catch (LuceneException | IllegalAccessException | IndexerRunningException ex) {
-            return new ResponseEntity<String>(ex.getMessage(), HttpStatus.BAD_REQUEST);
+        } catch (LuceneException | IllegalAccessException ex) {
+            return new ResponseEntity<String>(ex.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        } catch (IndexerRunningException e) {
+            return new ResponseEntity<String>(e.getMessage(), HttpStatus.SERVICE_UNAVAILABLE);
         }
 
         if (searchResults.length == 0) {
-            // Returning a 404 error.
+            // Data is not found but still returning OK
             return new ResponseEntity<String>("No records found for the search condition.",
```

Comment:
Let's return some proper xml here as well. Otherwise, it would throw clients off when they expect xml but just get text.